# Type: feat
# Subtype: Ex
# English Summary: Added new method `elementChanged()` to notify listeners when an element has been updated
# Chinese Summary: 增加新方法elementChanged()，用于通知监听器元素已更改
# Entry Number: 5

# === Git Diff Content ===

diff --git a/src/org/zaproxy/zap/utils/SortedComboBoxModel.java b/src/org/zaproxy/zap/utils/SortedComboBoxModel.java
index adda311d..d1ba35c0 100644
--- a/src/org/zaproxy/zap/utils/SortedComboBoxModel.java
+++ b/src/org/zaproxy/zap/utils/SortedComboBoxModel.java
@@ -67,4 +67,21 @@ public class SortedComboBoxModel<E extends Comparable<E>> extends DefaultComboBo
 	public void insertElementAt(E element, int index) {
 		addElement( element );
 	}
+
+	/**
+	 * Notifies the listeners that the given {@code element} of the combo box model was changed.
+	 * <p>
+	 * The call to this method has no effect if the given {@code element} doesn't exist in the combo box model.
+	 * </p>
+	 * 
+	 * @param element the element that was changed.
+	 * @since 2.3.0
+	 */
+	public void elementChanged(E element) {
+		int idx = getIndexOf(element);
+		if (idx < 0) {
+			return;
+		}
+		super.fireContentsChanged(this, idx, idx);
+	}
 }
